crates/subspace-core-primitives/src/objects.rs

```diff
@@ -172,3 +173,17 @@ impl GlobalObjectMapping {
         }
     }
 }
+
+/// Response to object mapping subscription, including a block height.
+/// Large responses are batched, so the block height can be repeated in different responses.
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
+pub struct ObjectMappingResponse {
+    /// The block number that the object mapping is from.
+    pub block_number: BlockNumber,
+
+    /// The object mappings.
+    #[cfg_attr(feature = "serde", serde(flatten))]
+    pub objects: GlobalObjectMapping,
+}
```

Comment:
I don't think "Response" belongs to the core primitives, this is very high-level much application level stuff. We have a separate crate for RPC primitives, would that be a better place for it?

Reply:
It’s used as an RPC response, then as an RPC and HTTP argument, so that makes sense. I’ll move it next time we change that code.

Reply:
Think of `subspace-core-protocol` as something third-party would have to have in order to build an alternative client. The bare minimum that is mandatory in all cases regardless of what the rest of the implementation looks like.

RPC stuff is much higher level than that, might look different or be missing altogether in some clients, etc.